src/retryOnEmpty.test.ts

```diff
@@ -750,19 +733,12 @@ function stubRequestThatFailsThenFinallySucceeds<
 }): ProviderRequestStub<T, U> {
   return stubGenericRequest({
     request,
-    response: (req, callNumber) => {
+    response: async (callNumber) => {
       if (callNumber <= numberOfTimesToFail) {
-        return {
-          id: req.id,
-          jsonrpc: '2.0',
-          error: {
-            code: -1,
-            message: 'oops',
-          },
-        };
+        throw new JsonRpcError(-1, 'oops');
```

Comment:
Nit: It would be more idiomatic to use the `custom` method on `providerErrors` (and you can get `providerErrors` from `@metamask/rpc-errors`):

``` suggestion
        throw providerErrors.custom({ code: -1, message: 'oops' });
```